tests/x509/limbo/test_limbo.py

```diff
@@ -0,0 +1,108 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+import datetime
+import json
+import os
+from ipaddress import IPv4Address
+
+from cryptography import x509
+from cryptography.x509 import load_pem_x509_certificate
+from cryptography.x509.verification import PolicyBuilder, Store
+
+LIMBO_UNSUPPORTED_FEATURES = {
+    # NOTE: Path validation is required to reject wildcards on public suffixes,
+    # however this isn't practical and most implementations make no attempt to
+    # comply with this.
+    "pedantic-public-suffix-wildcard",
+    # TODO: We don't support Distinguished Name Constraints yet.
+    "name-constraint-dn",
+    # Our support for custom EKUs is limited, and we (like most impls.) don't
+    # handle all EKU conditions under CABF.
+    "pedantic-webpki-eku",
+}
+
+
+def _get_limbo_peer(expected_peer, testcase_id):
+    assert expected_peer is not None, f"{testcase_id}: no expected peer name"
```

Comment:
We basically never use assertion messages. IME they're rarely useful compared to just using pdb + comments explaining what's going on

Reply:
af14ec426..eac3a0756